src/wp-includes/class-wp-script-modules.php

```diff
@@ -260,10 +278,12 @@ public function deregister( string $id ) {
 	 * @since 6.5.0
 	 */
 	public function add_hooks() {
-		$position = wp_is_block_theme() ? 'wp_head' : 'wp_footer';
-		add_action( $position, array( $this, 'print_import_map' ) );
-		add_action( $position, array( $this, 'print_enqueued_script_modules' ) );
-		add_action( $position, array( $this, 'print_script_module_preloads' ) );
+		add_action( wp_is_block_theme() ? 'wp_head' : 'wp_footer', array( $this, 'print_import_map' ) );
+		if ( wp_is_block_theme() ) {
```

Comment:
```suggestion
		if ( wp_is_block_theme() ) {
			// Modules can only be printed in the head for block themes because only with
			// block themes will import map be fully populated by modules discovered by
			// rendering the block template. In classic themes, modules are enqueued during
			// template rendering, thus the import map must be printed in the footer,
			// followed by all enqueued modules.
			add_action( 'wp_head', array( $this, 'print_head_enqueued_script_modules' ) );
		}
```